aspects/aspects.go

```diff
@@ -228,6 +237,91 @@ type accessPattern struct {
 	access accessType
 }
 
+// getPath returns a path if the specified name matches the access pattern's name
+// (with optional placeholders) and an empty string, if not.
+func (p *accessPattern) getPath(name string) (string, error) {
+	if strings.ContainsAny(p.name, "{}") {
+		matches := matchedPlaceholders(p.name, name)
+		if matches == nil {
+			// aspect name w/ placeholders doesn't match the specified name
+			return "", nil
+		}
+
+		return fillInPlaceholders(p.path, matches)
+	} else if p.name == name {
+		return p.path, nil
+	}
+
+	return "", nil
+}
+
+// matchedPlaceholders takes a dot-separated pattern with optional placeholders
+// and a name. To match the pattern, the name's dot-separated parts must be equal
+// to the pattern's non-placeholder parts (a pattern "foo.{bar}" is matched by a
+// name with a part "foo" followed by any part). It returns a map from placeholders
+// to their matches in the name (can be empty, if the path has no placeholders).
+// If there's no match, it returns nil.
+func matchedPlaceholders(pattern, name string) map[string]string {
+	patternParts, nameParts := strings.Split(pattern, "."), strings.Split(name, ".")
+
+	if len(patternParts) != len(nameParts) {
+		return nil
+	}
+
+	placeholderMatches := make(map[string]string)
+	for i, patt := range patternParts {
+		// TODO: check assertions for:
+		//	* empty name parts (e.g., "a..b")
+		//	* names with non-ASCII characters (or we can't index the name directly)
+		//	* mismatched or out-of-place curly brackets (assuming they can only be used for placeholders)
```

Comment:
I think these three points can be addressed by matching a regexp like `{[a-zA-Z0-9]+}`

Reply:
Thanks. Just to be clear, the validation of assertions is a TODO because we're passing them in hardcoded while working on the initial parts of aspects. But once we connect the aspects code to the other asserts machinery, this will become necessary. So this TODO just records assumptions regarding the aspect view names